tokio/src/sync/semaphore.rs

```diff
@@ -92,6 +92,11 @@ impl Semaphore {
     ///
     /// The maximum number of permits is `usize::MAX >> 3`, and this function will panic if the limit is exceeded.
     pub fn add_permits(&self, n: usize) {
+        assert!(
+            self.ll_sem.available_permits() + n <= ll::Semaphore::MAX_PERMITS,
+            "Reached more than max {} permits",
+            ll::Semaphore::MAX_PERMITS
+        );
```

Comment:
Why do you add it here when we already have an assert for this in `batch_semaphore.rs`?

Reply:
You're right. Actually, now that I removed it, I realized that in order to be consistent with the docs:
there was an assertion that wasn't including the upper bound.